### PR TITLE
refactor: moved untar to utils and added tests

### PR DESCRIPTION
**Pull request by mkanoor**
- Moved untar_project to utils.py as untar
- Added tests

### CODE REVIEW

File: ansible_rulebook/util.py [hunk 1]
```diff
@@ -202,3 +203,8 @@ def check_jvm():
             file=sys.stderr,
         )
         sys.exit(1)
+
+
+def untar(filename: str, output_dir: str) -> None:
```
Comment (Alex-Izquierdo) [inline ansible_rulebook/util.py:208]: why lose the asynchrony?

Comment (mkanoor) [reply ansible_rulebook/util.py:208]: @Alex-Izquierdo Not sure I haven't found a package that uses aiofile and tar file to do async processing. The earlier implementation was just using the asyncio.create_subprocess_exec to run the tar command directly.

Comment (mkanoor) [reply ansible_rulebook/util.py]: @Alex-Izquierdo I reverted to the old create_subprocess_exec.

**Review by Alex-Izquierdo — DISMISSED** [review]

**Comment by Alex-Izquierdo** [comment]
Can you update the install docs to mention the tar requirement?

**Review by Alex-Izquierdo — APPROVED** [review]

File: ansible_rulebook/builtin.py [hunk 2]
```diff
@@ -559,7 +542,10 @@ async def pre_process_runner(
     logger.debug("project_data_file: %s", project_data_file)
     if project_data_file:
         if os.path.exists(project_data_file):
-            await untar_project(project_dir, project_data_file)
+            rc = await untar(project_data_file, project_dir)
+            if rc != 0:
+                raise UntarFailedException("Untar of project file failed")
```
Comment (Alex-Izquierdo) [inline ansible_rulebook/builtin.py:547]: I think this logic should be contained in the function.

**Review by benthomasson — DISMISSED** [review]

**Review by Alex-Izquierdo — APPROVED** [review]

**Pull request closed without merging**